ts/src/hyperliquid.ts

```diff
@@ -1389,25 +1390,42 @@ export default class hyperliquid extends Exchange {
             const sz = this.amountToPrecision (symbol, amount);
             const reduceOnly = this.safeBool (orderParams, 'reduceOnly', false);
             const orderType: Dict = {};
-            if (isTrigger) {
+            if (isProtective) {
                 let isTp = false;
+                let tpslTriggerPrice = undefined;
                 if (takeProfitPrice !== undefined) {
-                    triggerPrice = this.priceToPrecision (symbol, takeProfitPrice);
+                    tpslTriggerPrice = this.priceToPrecision (symbol, takeProfitPrice);
                     isTp = true;
                 } else {
-                    triggerPrice = this.priceToPrecision (symbol, stopLossPrice);
+                    tpslTriggerPrice = this.priceToPrecision (symbol, stopLossPrice);
                 }
                 orderType['trigger'] = {
                     'isMarket': isMarket,
-                    'triggerPx': triggerPrice,
+                    'triggerPx': tpslTriggerPrice,
                     'tpsl': (isTp) ? 'tp' : 'sl',
                 };
+            } else if (isTrigger) {
+                const triggerDirection = this.safeString (orderParams, 'triggerDirection');
```

Comment:
@ttodua shouldn't we let the user also provide `tpsl` in `params` ?

Reply:
added support for it.